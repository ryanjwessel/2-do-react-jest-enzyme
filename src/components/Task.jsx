import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
const classNames = require('classnames');

// COMPONENTS
import TaskEditor from './TaskEditor.jsx';

class Task extends React.Component {
	constructor(props) {
		super(props);
		
		this.state = {
			editable: false,
		};
	}

	toggleEdit() {
		this.setState( { editable: !this.state.editable } );
	}

	toggleDone() {
		this.props.finishTask(this.props.task.id);
	}

	updateTask(task) {
		this.setState( {
			editable: !this.state.editable
		}, () => {
			this.props.updateTask(task);
		});
	}

	deleteTask() {
		this.props.deleteTask(this.props.task.id);
	}

	render() {
		const { task } = this.props;

		const classes = classNames(
			'col-xs-12',
			'task',
			{
				[ `task-${task.id}` ]: task.id,
				'done-task': task.done,
			},
		);

		return (
			<div className={ classes }>
				{ !this.state.editable ? (
					<div className="row middle-xs between-xs">
						<div className="col-xs-7 col-sm-9">
							<p className="toggle-task" onClick={() => this.toggleDone() }>{ task.name }</p>
							<p className="task-description">{ task.description }</p>
						</div>
						<div className="col-xs-5 col-sm-3">
							<button type="button" className="edit-task" onClick={() => this.toggleEdit() }>Edit</button>
							<button type="button" className="delete-task" onClick={() => this.deleteTask() }>Delete</button>
						</div>
					</div>
				) : (
					<Fragment>
						<TaskEditor
							task={ task }
							actionLabel="Update"
							handleNewTask={ (task) => this.updateTask(task) }
						/>
					</Fragment>
				) }
				<div className="col-xs-12">
					<hr />
				</div>
			</div>
		);
	}
}

Task.propTypes = {
	updateTask: PropTypes.func,
	finishTask: PropTypes.func,
	deleteTask: PropTypes.func,
	task: PropTypes.object,
};

export default Task;