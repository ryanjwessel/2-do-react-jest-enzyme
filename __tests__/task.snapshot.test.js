import Task from '../src/components/Task.jsx';
import renderer from 'react-test-renderer';
import React from 'react';

test('Task component renders the task correctly', () => {
	const task = { id: 1, done: false, name: 'Sample Task' };
	const rendered = renderer.create(
		<Task task={task} />
	);
	expect(rendered.toJSON()).toMatchSnapshot();
});
