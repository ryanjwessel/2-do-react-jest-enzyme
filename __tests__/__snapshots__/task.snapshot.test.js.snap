// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task component renders the task correctly 1`] = `
<div
  className="col-xs-12 task task-1"
>
  <div
    className="row middle-xs between-xs"
  >
    <div
      className="col-xs-7 col-sm-9"
    >
      <button
        className="toggle-btn add"
        onClick={[Function]}
        type="button"
      >
        âœ“
      </button>
      <span
        className="toggle-task"
      >
        Sample Task
      </span>
      <p
        className="task-description"
      />
    </div>
    <div
      className="col-xs-5 col-sm-3"
    >
      <button
        className="edit-task"
        onClick={[Function]}
        type="button"
      >
        Edit
      </button>
      <button
        className="delete-task"
        onClick={[Function]}
        type="button"
      >
        Delete
      </button>
    </div>
  </div>
  <div
    className="col-xs-12"
  >
    <hr />
  </div>
</div>
`;
